pages/login.tsx

```diff
@@ -0,0 +1,148 @@
+import axios from 'axios';
+import { GetServerSidePropsContext, GetServerSidePropsResult } from 'next';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { FormEvent, useState } from 'react';
+import { validateRequest } from '~/backend/auth';
+import { Button } from '~/components/Button';
+import { Input } from '~/components/Input';
+import { TitleText } from '~/components/TitleText';
+import SvgEyeOpen from '~/icons/eye_open';
+import SvgEyeSlash from '~/icons/eye_slash';
+import { isEmailValid } from '~/shared/isValidFunctions';
+import { User, UserLoginDetails } from '~/shared/types';
+import { handleLoginError } from '~/utils/handleError';
+
+export async function getServerSideProps(
+  context: GetServerSidePropsContext,
+): Promise<GetServerSidePropsResult<{ user: Partial<User> }>> {
+  const cookieData = await validateRequest(context.req, context.res);
+  if (!cookieData.user) {
+    return {
+      props: {
+        user: {},
+      },
```

Comment:
Tämän tarkoitus ei ole palauttaa mitään dataa, vaan ohjata käyttäjä pois `/login`-sivulta takaisin -> `/` sivulle, jos käyttäjä on jo authentikoitu

Tällä tyhjällä `user`-objektilla sain korjattua `ERR_TOO_MANY_REDIRECTS`-virheen :smile:

Reply:
Itsellä heitti seuraavan virheen: `Error: Your 'getServerSideProps' function did not return an object. Did you forget to add a 'return'`

Heittää siinä tilanteessa, kun ei ole kirjautunut sisään ja menee `/login`-sivulle

Poistin `user`-objektin tuosta palautuksesta, mutta `props`-objekti pitänee silti palauttaa. 

Eli tein tähän tapaan:
`if (!cookieData.user) {
    return {
      props: {},
    };
  }
`

En pysty tekemään ehdotusta, kun tämä kommentti on "vanha", siksi ei näy värejä yms :smile:

Reply:
Joo [`props`-objekti pitää palauttaa](https://nextjs.org/docs/pages/api-reference/functions/get-server-side-props#getserversideprops-return-values).